src/sentry/integrations/messaging.py

```diff
@@ -199,3 +225,283 @@ def build_handler(
         self, action: AlertRuleTriggerAction, incident: Incident, project: Project
     ) -> ActionHandler:
         return MessagingActionHandler(action, incident, project, self.spec)
+
+
+@control_silo_view
+class LinkingView(BaseView, ABC):
+    @property
+    @abstractmethod
+    def parent_messaging_spec(self) -> MessagingIntegrationSpec:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def provider(self) -> ExternalProviders:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def external_provider_enum(self) -> ExternalProviderEnum:
+        raise NotImplementedError
+
+    @property
+    def provider_slug(self) -> str:
+        return self.parent_messaging_spec.provider_slug
+
+    @property
+    @abstractmethod
+    def salt(self) -> str:
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def external_id_parameter(self) -> str:
+        raise NotImplementedError
+
+    # TODO: Replace thw two template properties below with base templates for all
+    #       integrations to use. Add service-specific parts to the context as needed.
+
+    @property
+    @abstractmethod
+    def confirmation_template(self) -> str:
+        """Path to the HTML template to render for a non-POST request."""
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def expired_link_template(self) -> str:
+        """Path to the HTML template to show when a link is expired."""
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_success_template_and_context(
+        self, params: Mapping[str, Any], integration: Integration | None
+    ) -> tuple[str, dict[str, Any]]:
+        """HTML content to show when the operation has been completed."""
+        raise NotImplementedError
+
+    @property
+    @abstractmethod
+    def metrics_operation_key(self) -> str:
+        raise NotImplementedError
+
+    def capture_metric(self, event_tag: str, tags: dict[str, str] | None = None) -> str:
+        event = ".".join(
+            ("sentry.integrations", self.provider_slug, self.metrics_operation_key, event_tag)
+        )
+        metrics.incr(event, tags=(tags or {}), sample_rate=1.0)
+        return event
+
+    @property
+    def has_analytics(self) -> bool:
+        return False
+
+    def record_analytic(self, event_tag: str, actor_id: int) -> None:
+        if not self.has_analytics:
+            return
```

Comment:
Previously only Discord had the `analytics.record` statement; this is to preserve that behavior without adding a new analytics key.

I'm not opposed to either adding it to the others or removing it from Discord, which would obviate the `has_analytics` property. But for now I'm trying to minimize behavior changes.

Reply:
I agree with minimizing the behavior change. But makes sense to have it in the parent class, can you please add a TODO here and also add it to the backlog you have for future work.